src/API/Authentication.php

```diff
@@ -231,120 +287,121 @@ public function oauth_token($options = [])
     /**
      * Makes a call to the `oauth/token` endpoint with `authorization_code` grant type.
      *
-     * @param $code
-     * @param $redirectUri
+     * @param string $code
+     * @param string $redirectUri
      *
-     * @return array|string
+     * @return array
      */
-    public function code_exchange($code, $redirectUri)
+    public function codeExchange($code, $redirectUri)
     {
         $options = [];
 
         $options['client_secret'] = $this->clientSecret;
         $options['redirect_uri'] = $redirectUri;
         $options['code'] = $code;
-        $options['grant_type'] = 'authorization_code';
 
-        return $this->oauth_token($options);
+        return $this->oauthToken('authorization_code', $options);
     }
 
     /**
      * Makes a call to the `oauth/token` endpoint with `password-realm` grant type.
      *
-     * @method login
+     * @param string $username
+     * @param string $password
+     * @param string $realm
+     * @param array $options {
+     *
+     *      @var string $scope    Optional
+     *      @var string $audience Optional.
+     * }
      *
-     * @param {Object} options:
-     * @param {Object} options.username
-     * @param {Object} options.password
-     * @param {Object} options.realm
-     * @param {Object} options.scope [optional]
-     * @param {Object} options.audience [optional]
+     * @return array
      */
-    public function login($options)
+    public function login($username, $password, $realm, array $options = [])
```

Comment:
So perhaps we should use this opportunity to promote some options to params, such as `scope`  and `audience`?